src/main/java/com/assylias/jbloomberg/DefaultBloombergSession.java

```diff
@@ -217,6 +210,7 @@ public synchronized void stop() {
             subscriptionManager.stop(this);
             session.stop();//started ? SYNC : ASYNC); //if not started, something's wrong, don't spend too much time here...
             state.set(TERMINATED);
+            updateStateListener();
```

Comment:
Please remove this line - the method will already be called from the EventHandler (actually adding that line breaks one of the tests).